packages/admin/admin/src/mui/menu/CollapsibleItem.tsx

```diff
@@ -46,53 +31,160 @@ const CollapsibleItem: React.FC<WithStyles<typeof styles> & MenuCollapsibleItemP
-    openedIcon = <ChevronUp />,
-    closedIcon = <ChevronDown />,
+    OpenedIcon,
+    ClosedIcon,
     children,
+    collapsedMenuTransitionComponent = Fade,
     ...otherProps
 }) => {
+    OpenedIcon = OpenedIcon || (isMenuOpen ? ChevronUp : ChevronRight);
+    ClosedIcon = ClosedIcon || (isMenuOpen ? ChevronDown : ChevronRight);
+
     const itemLevel: MenuLevel["level"] = level ? level : 1;
     let hasSelectedChild = false;
     const location = useLocation();
 
+    const [isCollapsibleOpen, setIsCollapsibleOpen] = React.useState<boolean>(openByDefault || hasSelectedChild);
+    const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null);
+
+    React.useEffect(() => {
+        // set open state manually to false to avoid a menu opening when isMenuOpen state changes
+        if (!isMenuOpen) setIsCollapsibleOpen(false);
+    }, [isMenuOpen]);
+
+    function checkIfPathInLocation(child: React.ReactElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>) {
+        return "to" in child.props && matchPath(location.pathname, { path: child.props.to, strict: true });
+    }
+
     const childElements = React.Children.map(children, (child: MenuChild) => {
-        if (matchPath(location.pathname, { path: child.props.to, strict: true })) {
+        // child is selected
+        if (checkIfPathInLocation(child)) {
+            hasSelectedChild = true;
+        }
+
+        // sub child is selected
+        const subChildElements =
+            "children" in child.props ? React.Children.map(child?.props?.children, (child: MenuChild) => child) : ([] as MenuChild[]);
+        if (subChildElements?.some((child: MenuChild) => child.props && checkIfPathInLocation(child))) {
             hasSelectedChild = true;
         }
 
         const newItemLevel = itemLevel + 1;
 
-        return React.cloneElement<MenuLevel>(child, {
+        return React.cloneElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>(child, {
             level: newItemLevel === 1 || newItemLevel === 2 || newItemLevel === 3 ? newItemLevel : undefined,
+            isMenuOpen,
         });
     });
 
-    const [open, setOpen] = React.useState<boolean>(openByDefault || hasSelectedChild);
+    const closeMenu = () => {
+        setAnchorEl(null);
+        setIsCollapsibleOpen(false);
+    };
+
+    const handlePopoverOpen = (event: React.MouseEvent<HTMLElement>) => {
+        if (isMenuOpen) return;
+        if (anchorEl !== event.currentTarget) {
+            setAnchorEl(event.currentTarget);
+            setIsCollapsibleOpen(true);
+        }
+    };
+
+    const handlePopoverClose = (e: React.MouseEvent<HTMLElement>) => {
+        if (isMenuOpen) return;
+        const el = e.target as HTMLElement;
+        const rect = el.getBoundingClientRect();
+        const { clientX, clientY } = e;
+
+        if (childElements?.length && clientX + 2 > rect.right && clientY > rect.top && clientY < rect.bottom) return;
```

Comment:
What's the idea behind this?

Reply:
This line checks if the mouse is over one of the child elements of the menu. If it is, the function returns and does nothing else. This is to prevent the menu from closing when the user is interacting with its contents.